### DISCUSSION TITLE
Guide: Setup Taskwarrior 3.0 server on Portainer

### DISCUSSION THREAD

**Discussion by zilto** [GothenburgBitFactory/taskwarrior#3709] [Show and tell]
Here's [a guide on my personal blog](https://tjean.me/til/2024/10/26/setup-taskwarrior-server) for setting up TaskWarrior 3.0 + the sync server in a container (via Portainer in my case). 

It wasn't too difficult, but the necessary information was scattered across several places so I decided to share my notes. 

## Main findings
- an official server image is available on GitHub
- the image hardcodes the path to mount as volume
- the client config fields changed between `<=3.1` and `>3.1`

## Thank you
Feel free to integrate that information in the README / docs if relevant. I can also open a PR if you'd like

Thanks for the work on TaskWarrior 3.0 ❤ I was very excited to setup the new sync server since never succeeded for 2.0.

**Comment by djmitche** [comment]
Regarding the config changing, the `origin` option still works for compatibility, it's just been removed from the documentation.

**Comment by djmitche** [comment]
I just added a docker-compose that will run taskwarrior-sync-server with a Caddy front-end, which does TLS using LetsEncrypt!
https://github.com/GothenburgBitFactory/taskchampion-sync-server/pull/68

**Reply by zilto** [reply]
Great, will take a look! For my home lab, I use Traefik, but it was a bigger headache to setup than expected 😅
